Instabile Tests bei AuthService

Im Rahmen von Update auf Version 0.2.0 viel auf das Test mal erfolgreich sind und mal Fehler werfen.

Hier der Auszug aus den Log des Workflows:

```log
2025-01-23T15:37:29.2130896Z [ERROR] Errors: 
2025-01-23T15:37:29.2134018Z [ERROR]   WahllokalBenutzerControllerIntegrationTest$CreateAndExportWahllokalBenutzer.should_deleteOldUsersWithWahltagIDAndPersistNewUsers_when_UsersAreGiven:126 » DataIntegrityViolation could not execute statement [Unique index or primary key violation: "PUBLIC.CONSTRAINT_INDEX_F ON PUBLIC.AUTHORITY(AUTHORITY NULLS FIRST) VALUES ( /* 3 */ 'WLS_USER_AUTHORITY_WAHLVORSTAND' )"; SQL statement:
2025-01-23T15:37:29.2137302Z insert into Authority (authority,id) values (?,?) [23505-224]] [insert into Authority (authority,id) values (?,?)]; SQL [insert into Authority (authority,id) values (?,?)]; constraint [PUBLIC.CONSTRAINT_INDEX_F]
2025-01-23T15:37:29.2141092Z [ERROR]   WahllokalBenutzerControllerIntegrationTest$CreateAndExportWahllokalBenutzer.should_persistUsers_when_usersAreGiven » DataIntegrityViolation could not execute statement [Referential integrity constraint violation: "CONSTRAINT_DF: PUBLIC.SECAUTHORITIES_SECPERMISSIONS FOREIGN KEY(PERMISSION_OID) REFERENCES PUBLIC.PERMISSION(ID) ('00000000-0000-0000-0001-000000000001')"; SQL statement:
2025-01-23T15:37:29.2145181Z delete from Permission where id=? [23503-224]] [delete from Permission where id=?]; SQL [delete from Permission where id=?]; constraint [CONSTRAINT_DF: PUBLIC.SECAUTHORITIES_SECPERMISSIONS FOREIGN KEY(PERMISSION_OID) REFERENCES PUBLIC.PERMISSION(ID) ('00000000-0000-0000-0001-000000000001'); SQL statement:
2025-01-23T15:37:29.2147086Z delete from Permission where id=? [23503-224]]
2025-01-23T15:37:29.2150134Z [ERROR]   WahllokalBenutzerControllerIntegrationTest$DeleteWahllokalBenutzer.should_removeOnlyUsersOfWahltag_when_wahltagIsGiven » DataIntegrityViolation could not execute statement [Referential integrity constraint violation: "CONSTRAINT_DF: PUBLIC.SECAUTHORITIES_SECPERMISSIONS FOREIGN KEY(PERMISSION_OID) REFERENCES PUBLIC.PERMISSION(ID) ('00000000-0000-0000-0001-000000000001')"; SQL statement:
2025-01-23T15:37:29.2155929Z delete from Permission where id=? [23503-224]] [delete from Permission where id=?]; SQL [delete from Permission where id=?]; constraint [CONSTRAINT_DF: PUBLIC.SECAUTHORITIES_SECPERMISSIONS FOREIGN KEY(PERMISSION_OID) REFERENCES PUBLIC.PERMISSION(ID) ('00000000-0000-0000-0001-000000000001'); SQL statement:
2025-01-23T15:37:29.2157711Z delete from Permission where id=? [23503-224]]
2025-01-23T15:37:29.2160844Z [ERROR]   WahllokalBenutzerControllerIntegrationTest$ExportWahllokalBenutzer.should_returnStringWithUsernames_when_usersFound » DataIntegrityViolation could not execute statement [Referential integrity constraint violation: "CONSTRAINT_DF: PUBLIC.SECAUTHORITIES_SECPERMISSIONS FOREIGN KEY(PERMISSION_OID) REFERENCES PUBLIC.PERMISSION(ID) ('00000000-0000-0000-0001-000000000001')"; SQL statement:
2025-01-23T15:37:29.2164997Z delete from Permission where id=? [23503-224]] [delete from Permission where id=?]; SQL [delete from Permission where id=?]; constraint [CONSTRAINT_DF: PUBLIC.SECAUTHORITIES_SECPERMISSIONS FOREIGN KEY(PERMISSION_OID) REFERENCES PUBLIC.PERMISSION(ID) ('00000000-0000-0000-0001-000000000001'); SQL statement:
2025-01-23T15:37:29.2166765Z delete from Permission where id=? [23503-224]]
2025-01-23T15:37:29.2167211Z [INFO] 
2025-01-23T15:37:29.2167561Z [ERROR] Tests run: 194, Failures: 0, Errors: 4, Skipped: 2
```

Ursache für das instabile Verhalten ist zu ermitteln und zu beheben.

--- a/wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/rest/UserControllerIntegrationTest.java
+++ b/wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/rest/UserControllerIntegrationTest.java
@@ -79,6 +79,7 @@ public class UserControllerIntegrationTest {
             userRepository.deleteUsersByWahltagID("wahltagID");
             authorityRepository.deleteAll();
             permissionRepository.deleteAll();
+            loginAttemptRepository.deleteAll();
         });
     }
 

--- a/wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/rest/WahllokalBenutzerControllerIntegrationTest.java
+++ b/wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/rest/WahllokalBenutzerControllerIntegrationTest.java
@@ -67,9 +67,11 @@ class WahllokalBenutzerControllerIntegrationTest {
 
     @AfterEach
     void tearDown() {
-        transactionTemplate.executeWithoutResult(status -> entityManager.createQuery("DELETE FROM User").executeUpdate());
-        permissionRepository.deleteAll();
-        authorityRepository.deleteAll();
+        transactionTemplate.executeWithoutResult(status -> {
+            entityManager.createQuery("DELETE FROM User").executeUpdate();
+            permissionRepository.deleteAll();
+            authorityRepository.deleteAll();
+        });
     }
 
     @Nested

